samples/ai-image-tagging/src/index.js

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import {
+  Button, SelectField, Option, Paragraph, FieldGroup, CheckboxField
+} from '@contentful/forma-36-react-components';
+import { init } from 'contentful-ui-extensions-sdk';
+import '@contentful/forma-36-react-components/dist/styles.css';
+import '@contentful/forma-36-fcss'
+
+import { ImageTaggingHelp } from './ImageTaggingHelp/ImageTaggingHelp'
+import { getAssetFields, findImageField, findTagsField, getTagFields } from "./lib/content-type";
+import { mergeTags, requestTags } from './lib/tags';
+
+import './index.css';
+
+class App extends React.Component {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      overwriteTags: true,
+      tagField: null,
+      imageField: null,
+    }
+  }
+
+  componentDidMount() {
+    this.props.sdk.window.startAutoResizer();
+
+    this.setState(state => ({
+      ...state,
+      tagField: findTagsField(this.props.sdk.contentType),
+      imageField: findImageField(this.props.sdk.contentType)
+    }))
+  }
+
+  loadTags = async () => {
+    const bail = (message) => {
+      notifier.error(message);
+      this.setState(state => ({
+        ...state,
+        loadingTags: false,
+      }))
+    };
+
+    this.setState(state => ({
+      ...state,
+      loadingTags: true,
+    }));
+
+    const { sdk } = this.props;
+    const { entry, space, notifier } = sdk;
+
+    const { imageField, tagField, overwriteTags } = this.state;
+
+    if (!imageField) {
+      bail('Add an image field to the content type first');
+      return
+    }
+
+    if (!tagField) {
+      bail('Add a text list field to the content type first');
+      return
+    }
+
+    let imageUrl;
+    try {
+      const assetId = entry.fields[imageField].getValue().sys.id;
+      const asset = await space.getAsset(assetId);
+      imageUrl = asset.fields.file['en-US'].url;
```

Comment:
Not sure on what you would add here, if there is no asset reference for the default locale it would fail with the message 'Add an image to the entry first'. We could explicitly tell the user, if there is more than one active locale, that we mean the field instance for the default locale.

Reply:
As discussed, already handled.